src/gadgets/CensorshipResistanceGadget.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity ^0.8.19;
+
+import {AttestationGadget} from "src/gadgets/AttestationGadget.sol";
+
+import {IAndromeda} from "src/IAndromeda.sol";
+
+abstract contract CensorshipResistanceGadget is AttestationGadget {
+    function Suave() internal virtual returns (IAndromeda);
+
+    // require explicit onchain challenge on kettle restart
+    mapping(bytes32 => bool) private cr_challenges;
+
+    function offchain_restart_challenge() public returns (bytes32 challenge, bytes memory attestation) {
+        require(!cr_challenges[challenge]); /* already challenged */
```

Comment:
this check seems off a bit.
It is checking if the _returned_ challenge was already done but it has not been set yet. So this require will always pass because challenge at this point is still empty and not set. right? or Did I oversee something?